ops/webidl/mod.rs

```diff
@@ -0,0 +1,246 @@
+                let val = ::deno_core::webidl::WebIdlConverter::convert(
+                  __scope,
+                  __value,
+                  __prefix.clone(),
+                  format!("'{}' of '{}' ({__context})", #string_name, #ident_string).into(),
+                )?;
+                #val
+              } else {
+                #required_or_default
+              }
+            };
+          }
+        }).collect::<Vec<_>>();
+
+        quote! {
+          ::deno_core::v8_static_strings! {
+            #(#v8_static_strings)*,
+          }
+
+          impl<'a> ::deno_core::webidl::WebIdlConverter<'a> for #ident {
+            fn convert(
+              __scope: &mut::deno_core:: v8::HandleScope<'a>,
+              __value: ::deno_core::v8::Local<'a, ::deno_core::v8::Value>,
+              __prefix: std::borrow::Cow<'static, str>,
+              __context: std::borrow::Cow<'static, str>
+            ) -> Result<Self, ::deno_core::webidl::WebIdlError> {
+              let __obj = if __value.is_undefined() || __value.is_null() {
+                None
+              } else {
+                Some(__value.to_object(__scope).ok_or_else(|| {
+                  ::deno_core::webidl::WebIdlError::new(
+                    __prefix.clone(),
+                    __context.clone(),
+                    ::deno_core::webidl::WebIdlErrorKind::ConvertToConverterType("dictionary")
+                  )
+                })?)
+              };
+
+              #(#fields)*
+
+              Ok(Self { #(#names),* })
+            }
+          }
+        }
+      }
+    },
+    Data::Enum(_) => {
+      return Err(Error::new(span, "Enums are currently not supported"));
+    }
+    Data::Union(_) => {
+      return Err(Error::new(span, "Unions are currently not supported"))
+    }
+  };
+
+  Ok(out)
+}
+
+mod kw {
+  syn::custom_keyword!(dictionary);
+}
+
+enum ConverterType {
+  Dictionary,
+}
+
+impl ConverterType {
+  fn from_attribute(attr: Attribute) -> Result<Option<Self>, Error> {
+    if attr.path().is_ident("webidl") {
+      let list = attr.meta.require_list()?;
+      let value = list.parse_args::<Self>()?;
+      Ok(Some(value))
+    } else {
+      Ok(None)
+    }
+  }
+}
+
+impl Parse for ConverterType {
+  fn parse(input: ParseStream) -> syn::Result<Self> {
+    let lookahead = input.lookahead1();
+
+    if lookahead.peek(kw::dictionary) {
+      input.parse::<kw::dictionary>()?;
+      Ok(Self::Dictionary)
+    } else {
+      Err(lookahead.error())
+    }
+  }
+}
+
+struct DictionaryField {
+  name: Ident,
+  default_value: Option<Expr>,
+  required: bool,
+}
+
+impl TryFrom<Field> for DictionaryField {
+  type Error = Error;
+  fn try_from(value: Field) -> Result<Self, Self::Error> {
+    let is_optional = if let Type::Path(path) = value.ty {
+      if let Some(last) = path.path.segments.last() {
+        last.ident == "Option"
+      } else {
+        false
+      }
+    } else {
+      false
+    };
```

Comment:
I don't know if this will work. We can have a field that is initialized to `undefined` but still required.

Reply:
good point. will rethink this

Reply:
maybe having a required attribute that only works on options might do the trick